bolt-modules/prompt/lib/puppet/functions/prompt.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'bolt/error'
+
+# Display a prompt and wait for a response.
+#
+# > **Note:** Not available in apply block
+Puppet::Functions.create_function(:prompt) do
+  # Wait until targets are available.
+  # @param prompt The prompt to display.
+  # @param options A hash of additional options.
+  # @option options [Boolean] sensitive Disable echo back and mark the response as sensitive.
+  # @return The response to the prompt.
+  # @example Prompt the user if plan execution should continue
+  #   $response = prompt('Continue executing plan? [Y\N]')
+  # @example Prompt the user for sensitive information
+  #   $password = prompt('Enter your password', 'sensitive' => true)
+  dispatch :prompt do
+    optional_param 'String[1]', :prompt
```

Comment:
I think it's reasonable to make this required. This is the `prompt` function, so one would naturally expect that you must provide a `prompt`. We can make it `String` instead of `String[1]` though to allow for empty prompts. It's also odd that the default value (`''`) isn't valid based on this type specification.

Reply:
Didn't realize that `[1]` was a parameter to the type... time to look over the types docs. :)